src/Shared/FileUtilities.cs

```diff
@@ -393,6 +398,48 @@ private static bool IsUNCPath(string path)
             return isUNC || path.IndexOf(@"\\?\globalroot", StringComparison.OrdinalIgnoreCase) != -1;
         }
 
+#if !CLR2COMPATIBILITY
+        /// <summary>
+        /// Gets the exact case used on the file system for an existing file or directory.
+        /// </summary>
+        /// <param name="path">A relative or absolute path.</param>
+        /// <returns>The full path using the correct case if the path exists. Otherwise, the full path.</returns>
+        private static string GetExactPath(string path)
+        {
+            // DirectoryInfo accepts either a file path or a directory path, and most of its properties work for either.
+            // However, its Exists property only works for a directory path, so we need an explicit file existence check.
+            var directory = new DirectoryInfo(path);
+            if (!File.Exists(path) && !directory.Exists)
```

Comment:
Need to propagate the EvaluationContext here to cache these calls.

Reply:
`EvaluationContext` isn't available here; it's actually prior to evaluation.

Reply:
The command line app can pass in a new context. But when this is called from an evaluation, it would need to get that evaluation's context. AFAIR the context is not far, maybe 2-3 frames up the stack. Wonder if the context would alleviate the perf tests a bit :)